pkg/gameserverallocations/allocator.go

```diff
@@ -607,6 +607,27 @@ func (c *Allocator) applyAllocationToGameServer(ctx context.Context, mp allocati
 	gs.ObjectMeta.Annotations[LastAllocatedAnnotationKey] = string(ts)
 	gs.Status.State = agonesv1.GameServerStateAllocated
 
+	// perfom any Counter or List actions
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		actionErrors := []string{}
+		if gsa.Spec.Counters != nil {
+			for counter, ca := range gsa.Spec.Counters {
+				errs := ca.CounterActions(counter, gs)
+				actionErrors = append(actionErrors, errs...)
+			}
+		}
+		if gsa.Spec.Lists != nil {
+			for list, la := range gsa.Spec.Lists {
+				errs := la.ListActions(list, gs)
+				actionErrors = append(actionErrors, errs...)
+			}
+		}
+		// record any Counter or List action errors as a warning
+		if len(actionErrors) != 0 {
+			c.recorder.Event(gs, corev1.EventTypeWarning, string(runtime.FeatureCountsAndLists), fmt.Sprint(actionErrors))
```

Comment:
A few thoughts:

* I think if we make `CounterActions` and `ListActions`  return a single either wrapped / multi -  error, this code becomes much simpler.
* Let's maybe have one potential warning for count actions and one for list actions? Seems like that would be clearer something like `c.recorder.Event(gs, corev1.EventTypeWarning, "CountAction", fmt.Sprint(actionErrors))` ??? WDYT?
* Rule of thumb on events: do them after the Update. The Update could fail for loads of reasons, so there's no point in doing the event until after the fact (events also are only create, so they are hard to fail on their own).

Reply:
Agreed and agreed for points 1 & 2.
Per your third point it makes sense to record any "warning" error after the Update, so I'll move this logic into allocationUpdateWorkers which is where we determine whether or not an Update is successful.

Reply:
Per our conversation moving event recording from allocationUpdateWorkers into applyAllocationToGameServer so we can record events after a successful udpate.